project/third_party/java/auto/BUILD

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2018 The Google Bazel Common Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# BUILD rules for https://github.com/google/auto
+
+load("@rules_java//java:defs.bzl", "java_library", "java_plugin")
+
+package(default_visibility = ["//visibility:public"])
+
+java_library(
+    name = "common",
+    exports = ["@com_google_auto_auto_common//jar"],
+)
+
+java_plugin(
+    name = "auto_value_processor",
+    processor_class = "com.google.auto.value.processor.AutoValueProcessor",
+    visibility = ["//visibility:private"],
+    deps = [
+        ":common",
+        ":service",
+        "//third_party/java/guava",
+        "@com_google_auto_value_auto_value//jar",
+    ],
+)
+
+java_plugin(
+    name = "auto_annotation_processor",
+    processor_class = "com.google.auto.value.processor.AutoAnnotationProcessor",
+    visibility = ["//visibility:private"],
+    deps = [
+        ":common",
+        ":service",
+        "//third_party/java/guava",
+        "@com_google_auto_value_auto_value//jar",
+    ],
+)
+
+java_plugin(
+    name = "auto_oneof_processor",
+    processor_class = "com.google.auto.value.processor.AutoOneOfProcessor",
+    visibility = ["//visibility:private"],
+    deps = [
+        ":common",
+        ":service",
+        "//third_party/java/guava",
+        "@com_google_auto_value_auto_value//jar",
+    ],
+)
+
+java_library(
+    name = "value",
+    exported_plugins = [
+        ":auto_annotation_processor",
+        ":auto_oneof_processor",
+        ":auto_value_processor",
+    ],
+    tags = ["maven:compile_only"],
+    exports = [
+        "//third_party/java/jsr250_annotations",  # TODO(ronshapiro) Can this be removed?
+        "@com_google_auto_value_auto_value_annotations//jar",
+    ],
+)
+
+java_plugin(
+    name = "auto_factory_processor",
+    generates_api = 1,
+    processor_class = "com.google.auto.factory.processor.AutoFactoryProcessor",
+    visibility = ["//visibility:private"],
+    deps = [
+        ":common",
+        ":service",
+        "//third_party/java/google_java_format",
+        "//third_party/java/guava",
+        "//third_party/java/javapoet",
+        "@com_google_auto_factory_auto_factory//jar",
+    ],
+)
+
+java_library(
```

Comment:
Can you please remove the targets that are not currently being used from this PR?

I think it's good practice to include build dependencies in the same PR where they are first used.

Reply:
Fixed in followup commit.